gcloud-java-compute/src/main/java/com/google/gcloud/compute/SnapshotInfo.java

```diff
@@ -0,0 +1,484 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Snapshot;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine snapshot. Compute Engine allows you to take snapshots of your persistent
+ * disk and create new persistent disks from that snapshot. This can be useful for backing up data,
+ * recreating a persistent disk that might have been lost, or copying a persistent disk. Snapshots
+ * can be applied across persistent disk types.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/persistent-disks#snapshots">Use
+ *     persistent disk snapshots</a>
+ */
+public class SnapshotInfo implements Serializable {
+
+  static final Function<Snapshot, SnapshotInfo> FROM_PB_FUNCTION =
+      new Function<Snapshot, SnapshotInfo>() {
+        @Override
+        public SnapshotInfo apply(Snapshot pb) {
+          return SnapshotInfo.fromPb(pb);
+        }
+      };
+  static final Function<SnapshotInfo, Snapshot> TO_PB_FUNCTION =
+      new Function<SnapshotInfo, Snapshot>() {
+        @Override
+        public Snapshot apply(SnapshotInfo snapshot) {
+          return snapshot.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 1065513502131159769L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String id;
+  private final SnapshotId snapshotId;
+  private final Long creationTimestamp;
+  private final String description;
+  private final Status status;
+  private final Long diskSizeGb;
+  private final List<LicenseId> licenses;
+  private final DiskId sourceDisk;
+  private final String sourceDiskId;
+  private final Long storageBytes;
+  private final StorageBytesStatus storageBytesStatus;
+
+  /**
+   * The status of a Google Compute Engine snapshot. A snapshot can be used to create other
+   * resources, such as disks, only after the snapshot has been successfully created and the status
```

Comment:
Can a snapshot be used to create resources other than disks?  If not, we could simplify the sentence to be "A snapshot can be used to create disks only after..."